model/godleyIcon.cc

```diff
@@ -277,6 +277,9 @@ namespace minsky
               {
                 FlowCoef fc(table.cell(r,c).substr(start));
                 v.init=fc.str();
+                // ensure that flows used as intial conditions, which inherit their values from other vars, correctly initialise corresponding stock var. for ticket 1137.
+                auto vv=minsky().variableValues[VariableValue::valueIdFromScope(group.lock(),v.init)];
+                if (vv->lhs()) vv->init=str(vv->value());
```

Comment:
Again, setting the initialisation string the the current value of a variable is the wrong thing. Initialisation should happen a reset time, and computed from the values of the variables at that point in time.

Reply:
I spent many hours this weekend trying initValue(minsky().variableValues) in several places (equation.cc and godleyIcon.cc). The problem is that it reinitialises ALL the variable values and not just the ones I'm interested in (a flow var with a lhs()).

Should I replicate what you do in ConstantDAG::addEvalOps with initValue():

          result=tmpResult;
        result->init=value;
        ***result=result->initValue(values);**

in VariableDAG::addEvalOps? ConstantDAG has a "value" string member. Would it make sense to add an "init" string member to VariableDAG for this?

Reply:
I'm not entirely sure why that code is there, to be honest. ConstantDAGs may not be associated with variable values stashed in the variableValues map, but if they are, then they should be consistent.